src/domain/create-user.use-case.ts

```diff
@@ -0,0 +1,47 @@
+import { useMutation } from "@apollo/client";
+import { useNavigate } from "react-router-dom";
+import { CreateUserMutation } from "../data/mutation/create-user";
+
+export type UserInput = {
+  name: string;
+  phone: string;
+  birthDate: string;
+  email: string;
+  role: string;
+  password: string;
+};
+
+export type CreateUserHookResult = {
+  createUser: (userData: UserInput) => Promise<void>;
+  loading: boolean;
+};
+
+export const useCreateUser = (): CreateUserHookResult => {
+  const navigation = useNavigate();
+  const token = localStorage.getItem("token");
+
+  const [createUserMutation, { loading }] = useMutation(CreateUserMutation);
+
+  const createUser: CreateUserHookResult["createUser"] = async (userData) => {
```

Comment:
precisa colocar a tipagem dessa forma na variável, o ts não consegue inferir?
```suggestion
  const createUser: CreateUserHookResult["createUser"] = async (userData) => {
```

Reply:
Sim aliais o TypeScript pode inferir o tipo com base na assinatura da função.